tests/testthat/test-code_dependency.R

```diff
@@ -0,0 +1,414 @@
+      "a <- 1"
+    )
+  }
+)
+
+
+
+# functions -------------------------------------------------------------------------------------------------------
+
+testthat::test_that("get_code ignores occurrence in function definition", {
+  q <- new_qenv()
+  q <- eval_code(q, "b <- 2")
+  q <- eval_code(q, "foo <- function(b) { b <- b + 2 }")
+
+  testthat::expect_identical(
+    get_code(q, names = "foo"),
+    "foo <- function(b) {\n    b <- b + 2\n}"
+  )
+})
+
+
+testthat::test_that("get_code ignores effect of the object which occurs in a function definition", {
+  q <- new_qenv()
+  q <- eval_code(q, "b <- 2")
+  q <- eval_code(q, "foo <- function(b) { b <- b + 2 }")
+
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    c("b <- 2")
+  )
+})
+
+testthat::test_that("get_code detects occurrence of the function object", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- 2")
+  q <- eval_code(q, "foo <- function(b) { b <- b + 2 }")
+  q <- eval_code(q, "b <- foo(a)")
+
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    c("a <- 1", "b <- 2", "foo <- function(b) {\n    b <- b + 2\n}", "b <- foo(a)")
+  )
+})
+
+testthat::test_that(
+  "Can't detect occurrence of function definition when a formal is named the same as a function",
+  {
+    testthat::skip("This does not return foo definition YET!")
+    q <- new_qenv()
+    q <- eval_code(q, "x <- 1")
+    q <- eval_code(q, "foo <- function(foo = 1) 'text'")
+    q <- eval_code(q, "a <- foo(x)")
+
+    testthat::expect_identical(
+      get_code(q, names = "a"),
+      c("x <- 1", "foo <- function(foo = 1) 'text'", "a <- foo(x)")
+    )
+  }
+)
+
+# $ ---------------------------------------------------------------------------------------------------------------
+
+
+testthat::test_that("get_code understands $ usage and do not treat rhs of $ as objects (only lhs)", {
+  q <- new_qenv()
+  q <- eval_code(q, "x <- data.frame(a = 1:3)")
+  q <- eval_code(q, "a <- data.frame(y = 1:3)")
+  q <- eval_code(q, "a$x <- a$y")
+  q <- eval_code(q, "a$x <- a$x + 2")
+  q <- eval_code(q, "a$x <- x$a")
+
+  testthat::expect_identical(
+    get_code(q, names = "x"),
+    c("x <- data.frame(a = 1:3)")
+  )
+  testthat::expect_identical(
+    get_code(q, names = "a"),
+    c("x <- data.frame(a = 1:3)", "a <- data.frame(y = 1:3)", "a$x <- a$y", "a$x <- a$x + 2", "a$x <- x$a")
+  )
+})
+
+testthat::test_that("get_code detects cooccurrence properly even if all objects are on rhs", {
+  q <- new_qenv()
+  q <- eval_code(q, "a <- 1")
+  q <- eval_code(q, "b <- list(c = 2)")
+  q <- eval_code(q, "b[[a]] <- 3")
+
+  testthat::expect_identical(
+    get_code(q, names = "b"),
+    c("a <- 1", "b <- list(c = 2)", "b[[a]] <- 3")
+  )
+})
+
+
+# @ ---------------------------------------------------------------------------------------------------------------
+
+testthat::test_that("get_code understands @ usage and do not treat rhs of @ as objects (only lhs)", {
```

Comment:
hey @gogonzo ánd @chlebowa - pinging again
this test does not fail locally but fails in GHA https://github.com/insightsengineering/teal.code/actions/runs/6431343460/job/17464067656?pr=146#step:39:205

Reply:
1. `representation` is deprecated.
2. I suppose `representation` contains slot definitions. Those slot names are the same as (some) variable names. Could it be to do with that?

Reply:
omiting that since get_code_dependency is not connected at this stage to eval_code